### PR TITLE
testmo checks fails on community pull requests

### PR DESCRIPTION
**Pull request by wojteknowacki**
PR from the community has no access to secrets, making testmo steps fail. I added the condition to trigger testmo only if the secret is available.

### Pull Request Checklist

1. [ ] This code contains UI changes
2. [ ] All visible strings are translated with proper context including data-formatting
3. [ ] Attributes `data-test-id` are added for new elements
4. [ ] The changes are tested in Chrome/Firefox/Safari browsers and in light/dark mode
5. [ ] Your code works with the latest stable version of the core
6. [ ] I added changesets and [read good practices](/.changeset/README.md)

### Test environment config

API_URI=https://automation-dashboard.staging.saleor.cloud/graphql/
APPS_MARKETPLACE_API_URI=https://apps.staging.saleor.io/api/v2/saleor-apps

### Do you want to run more stable tests?

To run all tests, just select the stable checkbox. To speed up tests, increase the number of containers. Tests will be re-run only when the "run e2e" label is added.

1. [ ] stable
2. [ ] app
3. [ ] attribute
4. [ ] category
5. [ ] collection
6. [ ] customer
7. [ ] giftCard
8. [ ] homePage
9. [ ] login
10. [ ] menuNavigation
11. [ ] navigation
12. [ ] orders
13. [ ] pages
14. [ ] payments
15. [ ] permissions
16. [ ] plugins
17. [ ] productType
18. [ ] products
19. [ ] sales
20. [ ] shipping
21. [ ] translations
22. [ ] variants
23. [ ] vouchers

CONTAINERS=2

### CODE REVIEW

**Review by krzysztofzuraw — DISMISSED** [review]

File: .github/workflows/test-env-deploy.yml [hunk 1]
```diff
@@ -3,6 +3,16 @@ name: TEST-ENV-DEPLOYMENT
...
+  check-secret:
+    runs-on: ubuntu-22.04
+    outputs:
+      testmo-url: ${{ steps.testmo-url.outputs.defined }}
+    steps:
+      - id: testmo-url
+        env:
+          TESTMO_URL: ${{ secrets.TESTMO_URL }}
+        if: "${{ env.TESTMO_URL != '' }}"
+        run: echo "::set-output name=defined::true"
```
Comment (NyanKiyoshi) [inline .github/workflows/test-env-deploy.yml:15]: `::set-output` is deprecated https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/

File: .github/workflows/test-env-deploy.yml [hunk 2]
```diff
@@ -171,7 +181,8 @@ jobs:
           echo ${{steps.get_tags.outputs.result}}
 
   testmo-report-preparation:
-    needs: prepare-tests
+    needs: [check-secret, prepare-tests]
+    if: needs.check-secret.outputs.testmo-url == 'true'
```
Comment (NyanKiyoshi) [inline .github/workflows/test-env-deploy.yml:185]: Why not simply do this?

```
if: github.repository == 'saleor/saleor-dashboard'
```

Comment (wojteknowacki) [reply .github/workflows/test-env-deploy.yml:185]: Will that do the same when it comes to community PRs?

Comment (NyanKiyoshi) [reply .github/workflows/test-env-deploy.yml:185]: That's only for `push` event, I see it's a `pull_request` event, thus you could use:

```
github.repository == github.event.pull_request.head.repo.full_name
```

or

```
github.event.pull_request.head.repo.fork == false
```

**Review by NyanKiyoshi — APPROVED** [review]

**Review by krzysztofzuraw — APPROVED** [review]

**Review by NyanKiyoshi — APPROVED** [review]

**Review by krzysztofzuraw — APPROVED** [review]

**Pull request merged**